### ISSUE TITLE
Image data repeat resize and compute ratio bugs

### ISSUE THREAD

**Issue by OceanWong1991** [Megvii-BaseDetection/YOLOX#1747]
@FateScript @Joker316701882 Thanks for Sharing this great project!

I think we may do repeat image data resize and compute ratio

1. In [load_resized_img](https://github.com/Megvii-BaseDetection/YOLOX/blob/ac58e0a5e68e57454b7b9ac822aced493b553c53/yolox/data/datasets/coco.py#L133) and [load_anno_from_ids](https://github.com/Megvii-BaseDetection/YOLOX/blob/ac58e0a5e68e57454b7b9ac822aced493b553c53/yolox/data/datasets/coco.py#L92C17-L92C17) we do the resize image first time.
2. In [MosaicDetection](https://github.com/Megvii-BaseDetection/YOLOX/blob/ac58e0a5e68e57454b7b9ac822aced493b553c53/yolox/data/datasets/mosaicdetection.py#L96), We repeat do the image resize
3. In [TrainTransform](https://github.com/Megvii-BaseDetection/YOLOX/blob/ac58e0a5e68e57454b7b9ac822aced493b553c53/yolox/data/data_augment.py#L149), We do resize image again

Is that right?

**Comment by OceanWong1991** [comment]
@FateScript You can debug it, the ratio is constant 1.0, because when we load image and anno, we already do the resize and ratio work.

**Comment by OceanWong1991** [comment]
I also make a PR fix this bug, and past the test on my server, if there is anything wrong with my understanding, plz tell me, Thx~

**Comment by FateScript** [comment]
Thanks for you contribution : ) 
Actucally this code is writen by design. User could delete this code or write their own version. We add 3 resize code is to make sure those people who fork this repo will not doing things wrong if they modify the code a little.

**Comment by OceanWong1991** [comment]
@FateScript Thx for your replying. Actually, I DO NOT think repeat write same code is a good to prevent users from making mistakes.

**Comment by FateScript** [comment]
Yes, I totally agree with you. In fact, the root reason for this is that we are lack of time when we prepara to release the code. Dataloader and its related code in this repo should be refactored actually.
